app/celer-g4/DetectorConstruction.cc

```diff
@@ -139,6 +186,22 @@ void DetectorConstruction::ConstructSDandField()
         // Hand SD to the manager
         sd_manager->AddNewDetector(detector.release());
     }
+
+    // Construct the magnetic field
+    G4FieldManager* field_manager
+        = G4TransportationManager::GetTransportationManager()->GetFieldManager();
+
+    if (use_field_map_)
+    {
+        mag_field_ = new MagneticField(field_params_);
+    }
+    else
+    {
+        G4ThreeVector field = GlobalSetup::Instance()->GetMagFieldZTesla();
+        mag_field_ = new G4UniformMagField(field);
+    }
```

Comment:
What's the need to have a static thread-local `mag_field_` when it's only ever used locally in this function? Can't it just be created locally? (If the field manager doesn't take ownership, should we be using a shared_ptr?) Since the field itself should be the same on all threads, why do we have to recreate it for each thread?

Reply:
Thanks for pointing and now using shared_ptr.